format-avro/src/main/java/io/cdap/plugin/format/avro/output/GenericRecordWrapper.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.avro.output;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+
+/**
+ * TODO: add
+ */
+public class GenericRecordWrapper {
+  private final GenericRecord genericRecord;
+  private final Schema schema;
+  private final int hash;
+
+  public GenericRecordWrapper(GenericRecord genericRecord, Schema schema, int hash) {
```

Comment:
I use the Hash for uniqueness (only one writer per distinct schema) and the Schema entity is used to lazily initialize the writer.